src/main.rs

```diff
@@ -431,12 +441,63 @@ fn build_ui(app: &App, application: &gtk::Application) {
     main_menu_model.append("About", "app.about");
     main_menu.set_menu_model(&main_menu_model);
 
+    let overlay = gtk::Overlay::new();
+    let overlay_text = gtk::Label::new("0");
+
+    overlay.add_overlay(&overlay_text);
+    overlay_text.set_halign(gtk::Align::End);
+    overlay_text.set_valign(gtk::Align::Start);
+    if let Some(style) = overlay_text.get_style_context() {
+        let css_provider = gtk::CssProvider::new();
+        css_provider.load_from_data("GtkLabel {
+            background-color: #fff;
+            border: 1px solid black;
+            border-radius: 2px;
+            color: black;
+        }".as_bytes()).expect("load_from_data failed");
+        style.add_provider(&css_provider, gtk::STYLE_PROVIDER_PRIORITY_USER);
+    }
+    overlay_text.set_no_show_all(true);
+    overlay_text.set_visible(false);
+
     let snapshot_button = gtk::Button::new();
     let snapshot_button_image = gtk::Image::new_from_icon_name("camera-photo", 1);
     snapshot_button.add(&snapshot_button_image);
 
     snapshot_button.connect_clicked(move |_| {
-        // TODO: call gstreamer snapshot
+        let settings = load_settings();
+
+        if let Some(t) = timeout.borrow_mut().take() {
+            glib::source::source_remove(t);
+            overlay_text.set_visible(false);
+            return
+        }
+        if settings.timer_length == 0 {
+            take_snapshot();
+        } else {
+            overlay_text.set_visible(true);
```

Comment:
It goes into this branch for me but I never actually see the overlay

Reply:
Ah? That is surprising... It works well for me (on linux mint).

Reply:
Just set it visible all the time. Like that you'll know for sure it's not something that isn't working.

Reply:
Doesn't appear then either :)

Reply:
... Do the overlay example works for you? (the C one in GNOME example repository)

Reply:
The ones from `gtk3-demo`? Yes those work for me. I'll debug this later :)

Linux Mint means no gnome-shell and no Wayland I assume?

Reply:
There is gnome-shell apparently but not Wayland (that might it?).